sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -532,6 +534,61 @@ object DateTimeUtils extends SparkDateTimeUtils {
     }
   }
 
+  /**
+   * Truncate a TIME-of-day value given as microseconds from midnight.
+   *
+   * @param unitUtf8  the requested unit ('HOUR', 'MINUTE', ...) or null
+   * @param micros microseconds since 00:00:00
+   * @return truncated value as microseconds-of-day, or -1L if the unit is invalid
+   */
+  def truncateTime(unitUtf8: UTF8String, micros: Long): Long = {
+    val level = parseTruncLevelForTime(unitUtf8)
+    if (level < MIN_LEVEL_OF_TIME_TRUNC) {  // unknown unit
+      -1L
+    } else {
+      val out = truncTime(micros, level)         // may still return -1L on error
```

Comment:
What's the reason of splitting the function? Is `truncTime` used somewhere else?